syntax = "proto3";
import "google/protobuf/empty.proto";

// to compile
//python3 -m grpc_tools.protoc -I./ --python_out=. --grpc_python_out=. --pyi_out=. ./meteo_utils.proto

package meteo;

message AirConditionParams {
  float temperature = 1;
  float co2 = 2;
  float humidity = 3;
}

message AirAnalysisResponse {
    float temperature = 1;
    float humidity = 2;
}
message PollutionAnalysisResponse {
    float co2 = 1;
}

message AirWellness {
  float wellness = 1;
}

message Co2Wellness {
  float wellness = 1;
}


service MeteoDataService {

  rpc ProcessMeteoData(AirAnalysisResponse) returns (AirWellness) {}
  rpc ProcessPollutionData(PollutionAnalysisResponse) returns (Co2Wellness) {}
  rpc AnalyzeAir(google.protobuf.Empty) returns (AirAnalysisResponse) {}
  rpc AnalyzePollution(google.protobuf.Empty) returns (PollutionAnalysisResponse) {}
}

